### PR TITLE
Bugfix/fix unsupported exception

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radixdlt-java/src/main/java/com/radixdlt/client/core/network/RadixNetworkController.java [hunk 1]
```diff
@@ -120,8 +120,14 @@ private RadixNetworkController(
 			.map(epic -> epic.epic(connectableReducedNodeActions, networkState))
 			.collect(Collectors.toSet());
 
-		// FIXME: Cleanup disposable
-		Observable.merge(updates).subscribe(this::dispatch);
+		// FIXME: Cleanup disposable and error handling
```
Comment (laszlourszuly) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/network/RadixNetworkController.java]: Is this `FIXME` still valid?

File: radixdlt-java/src/main/java/com/radixdlt/client/core/network/RadixNetworkController.java [hunk 2]
```diff
@@ -120,8 +120,14 @@ private RadixNetworkController(
...
+		Observable.merge(updates).subscribe(
+			this::dispatch,
+			e -> {
+				e.printStackTrace();
```
Comment (laszlourszuly) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/network/RadixNetworkController.java]: Do we have a dedicated logger (like `slf4j` or similar) that could/would/should handle the printing of this exception instead? (Benefit being that the logger can be configured "globally" to control output to some degree, while `printStackTrace()` doesn't honour any of those configurations).

File: radixdlt-java/src/main/java/com/radixdlt/client/core/network/actions/DiscoverMoreNodesErrorAction.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+	public Throwable getReason() {
+		return reason;
+	}
+
+	// TODO: remove this
+	@Override
+	public RadixNode getNode() {
+		throw new UnsupportedOperationException();
+	}
```
Comment (laszlourszuly) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/network/actions/DiscoverMoreNodesErrorAction.java]: Oh, the ruthless humour of the Universe 😂

File: radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/DiscoverNodesEpic.java [hunk 4]
```diff
@@ -34,9 +35,9 @@ public DiscoverNodesEpic(Observable<RadixNode> seeds, RadixUniverseConfig config
 	public Observable<RadixNodeAction> epic(Observable<RadixNodeAction> updates, Observable<RadixNetworkState> networkState) {
 		Observable<RadixNodeAction> getSeedUniverses = updates
 			.ofType(DiscoverMoreNodesAction.class)
-			.firstOrError()
-			.flatMapObservable(i -> seeds)
-			.map(GetUniverseRequestAction::of);
+			.flatMap(i -> seeds)
+			.<RadixNodeAction>map(GetUniverseRequestAction::of)
+			.onErrorReturn(DiscoverMoreNodesErrorAction::new);
```
Comment (laszlourszuly) [inline radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/DiscoverNodesEpic.java]: Is this `onErrorReturn(...)` for not breaking the observable stream (as a hard `onError` would) and thereby saving me the effort of having to "reconnect" after each error?

Beautiful, if so!

File: radixdlt-java/src/main/java/com/radixdlt/client/core/network/RadixNetworkController.java (hunk 1 above)
Comment (talekhinezh) [reply radixdlt-java/src/main/java/com/radixdlt/client/core/network/RadixNetworkController.java]: Yes, should really not do a `subscribe` inside constructor but it has been good enough for now.

File: radixdlt-java/src/main/java/com/radixdlt/client/core/network/RadixNetworkController.java (hunk 2 above)
Comment (talekhinezh) [reply radixdlt-java/src/main/java/com/radixdlt/client/core/network/RadixNetworkController.java]: good point

File: radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/DiscoverNodesEpic.java (hunk 4 above)
Comment (talekhinezh) [reply radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/DiscoverNodesEpic.java]: Yup, exactly. It's RX version of try/catch.

**Pull request merged**